### PR TITLE
feat: Shift the jql construction to the user

### PR DESCRIPTION
**Pull request by sgkiokas**
This PR:
- shifts the jql consutrction is shifted to the user by letting him specify the criteria of how to search of resolved and open issues as
part of the action's duplication mechanism prevention
- fixes a bug with the UPLOAD_FILES parameter and with the search criteria of the issues that are already open
- updates the unit tests

Contributes to: cameltols/actions-jira-integration#126

Signed-off-by: Stelios Gkiokas <s_giokas@hotmail.com>

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -92,9 +91,10 @@ const kickOffAction = async (inputJson) => {
     Object.entries(utils.populateMap(PRIORITY_MAPPER))
   );
   const reportPairsMapper = utils.populateMap(REPORT_INPUT_KEYS);
+  const issueLabelsMapper = utils.getInput('ISSUE_LABELS_MAPPER');
   const labels =
-    ISSUE_LABELS_MAPPER.length !== 0
-      ? { labels: ISSUE_LABELS_MAPPER.split(',') }
```
Comment (konstantinos-perperidis) [inline index.js:97]: What if length is exactly === 1? Doesn't if matter to Jira to have one entry in the labels list in the query with a trailing comma?

**Review by konstantinos-perperidis — CHANGES_REQUESTED** [review]
Left one comment.

File: index.js (hunk 1 above)
Comment (sgkiokas) [reply index.js:97]: Yep, this is a valid point. Will address.

**Review by konstantinos-perperidis — APPROVED** [review]
LGTM.

**Pull request merged**